go/service/gregor.go

```diff
@@ -1391,19 +1418,21 @@ type timeoutClient struct {
 var _ rpc.GenericClient = (*timeoutClient)(nil)
 
 func (t *timeoutClient) Call(ctx context.Context, method string, arg interface{}, res interface{}) error {
-	timeoutCtx, timeoutCancel := context.WithTimeout(ctx, t.timeout)
+	var timeoutCancel context.CancelFunc
+	ctx, timeoutCancel = context.WithTimeout(ctx, t.timeout)
```

Comment:
Is reassigning to the `ctx` variable a deliberate change here, or mostly left over from the previous version of the diff? We might be able to undo all the changes down in this part of the file?